src/CookieCollection.php

```diff
@@ -0,0 +1,352 @@
+     *
+     * @return int the number of cookies in the collection.
+     */
+    public function count(): int
+    {
+        return count($this->cookies);
+    }
+
+    /**
+     * Returns the cookie with the specified name.
+     *
+     * @param string $name the cookie name
+     * @return Cookie the cookie with the specified name. Null if the named cookie does not exist.
+     * @see getValue()
+     */
+    public function get(string $name): Cookie
+    {
+        return $this->cookies[$name];
+    }
+
+    /**
+     * Returns the value of the named cookie.
+     *
+     * @param string $name the cookie name
+     * @param mixed $defaultValue the value that should be returned when the named cookie does not exist.
+     * @return string|null the value of the named cookie or the default value if cookie is not set.
+     * @see get()
+     */
+    public function getValue(string $name, $defaultValue = null): ?string
+    {
+        return isset($this->cookies[$name]) ? $this->cookies[$name]->getValue() : $defaultValue;
+    }
+
+    /**
+     * Adds a cookie to the collection.
+     * If there is already a cookie with the same name in the collection, it will be removed first.
+     *
+     * @param Cookie $cookie the cookie to be added
+     */
+    public function add(Cookie $cookie): void
+    {
+        $this->cookies[$cookie->getName()] = $cookie;
+    }
+
+    /**
+     * Returns whether there is a cookie with the specified name.
+     *
+     * @param string $name the cookie name
+     * @return bool whether the named cookie exists
+     * @see remove()
+     */
+    public function has(string $name): bool
+    {
+        return isset($this->cookies[$name]);
+    }
+
+    /**
+     * Removes a cookie.
+     *
+     * @param string $name the name of the cookie to be removed.
+     * @return Cookie|null cookie that is removed
+     */
+    public function remove(string $name): ?Cookie
+    {
+        if (!isset($this->cookies[$name])) {
+            return null;
+        }
+
+        $removed = $this->cookies[$name];
+        unset($this->cookies[$name]);
+
+        return $removed;
+    }
+
+    /**
+     * Set cookie expire date to outdated to further remove it from browser.
+     *
+     * @param string $name the name of the cookie to expire.
+     */
+    public function expire(string $name): void
+    {
+        if (!isset($this->cookies[$name])) {
+            return;
+        }
+
+        $this->cookies[$name] = $this->cookies[$name]->withExpires(new DateTimeImmutable('-1 year'));
+    }
+
+    /**
+     * Removes all cookies.
+     */
+    public function clear(): void
+    {
+        $this->cookies = [];
+    }
+
+    /**
+     * Returns whether the collection already contains the cookie.
+     *
+     * @param Cookie $cookie the cookie to check for
+     * @return bool whether cookie exists
+     * @see has()
+     */
+    public function contains(Cookie $cookie): bool
```

Comment:
Isn't it the same as `has()`?

Reply:
Nope, `has()` just checks the existence by cookie name, `contains` checks the existence by value (or in this case by reference).

Also implementation of the planned interface (see answer below).